org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/computation/CollectionLiteralsTypeComputer.java

```diff
@@ -176,7 +177,17 @@ protected void computeType(XListLiteral literal, JvmGenericType listType, ITypeE
 
 	protected void computeArrayLiteralType(XListLiteral literal, LightweightTypeReference expectedArrayType, ITypeExpectation expectation,
 			ITypeComputationState state) {
-		LightweightTypeReference elementTypeExpectation = expectedArrayType.getComponentType();
+		LightweightTypeReference elementTypeExpectation = null;
```

Comment:
@cdietrich good question, as far as i see at least in the case of a synonym compound reference with array components it should work